pkg/fs/fs-bucket-listobjects.go

```diff
@@ -64,7 +64,12 @@ func (fs Filesystem) ListObjects(bucket, prefix, marker, delimiter string, maxKe
 		}
 	}
 
-	if maxKeys <= 0 || maxKeys > listObjectsLimit {
+	// Return empty resposne for a valid request when maxKeys is 0.
+	if maxKeys == 0 {
+		return result, nil
+	}
+
+	if maxKeys < 0 || maxKeys > listObjectsLimit {
 		maxKeys = listObjectsLimit
 	}
```

Comment:
If you already have tests in another PR its is OK. Your case is an exception, since you own the tests.